app/shared/env/env-check.js

```diff
@@ -0,0 +1,24 @@
+// @ts-check
+/**
+ * This file is included in `/remix.config.js` which ensures the app isn't built with invalid env vars.
+ * It has to be a `.js`-file to be imported there.
+ */
+/* eslint-disable @typescript-eslint/no-var-requires */
+const { z } = require("zod");
+
+const envSchema = z.object({
+  DATABASE_URL: z.string().url(),
+  AUTH_SECRET: z.string().min(10),
```

Comment:
is this always required or only in dev mode?

Reply:
This will also work on build

Reply:
If you don't have the vars the build will break

Reply:
Sooo, researched this and in production remix doesn't read the .env file so this is only for dev

Reply:
that means that we need a place that is going to be executed for example during deployment to production, because you want a good error message if the env wasn't configured correctly and it would run in both dev and prod

Reply:
I guess in prod you want to catch it before its in prod so that deployment is canceled

Reply:
has been fixed below